lib/charms/traefik_k8s/v0/ingress_per_unit.py

```diff
@@ -76,6 +75,20 @@ def _handle_ingress_per_unit(self, event):
 
 log = logging.getLogger(__name__)
 
+try:
+    import jsonschema
+
+    DO_VALIDATION = True
+except ModuleNotFoundError:
+    log.warning(
+        "The `ingress_per_unit` library needs the `jsonschema` package to be able "
+        "to do runtime data validation; without it, it will still work but validation "
+        "will be disabled. \n"
+        "It is recommended to add `jsonschema` to the 'requirements.txt' of your charm, "
+        "which will enable this feature."
+    )
+    DO_VALIDATION = False
```

Comment:
Do we need to surface this to the charm dev in any way or do we feel that a warning in the debug-log is enough?

Reply:
Well it depends on how essential we think validation should be to this library. If we want it to work without it, and validation is an opt-in additional feature, then this is fine.
On the other hand I can imagine there can be a situation where 
1) person uses lib without jsonschema, ignores the warnings
2) person installs jsonschema in charm for unrelated reasons
3) charm starts breaking because validation is suddenly active and the schema is wrong/outdated
4) person is puzzled

@jnsgruk @mmanciop what is your view on this? 
I recently merged a pr from @mmanciop, his take was less permissive: install jsonschema or get an Exception.

Reply:
FWIW it's still pretty obvious that there's something wrong:
![image](https://user-images.githubusercontent.com/6230162/163196924-5ea16a07-cecf-4610-b3c7-37960f6652f7.png)

Reply:
I'm happy with where it's at - the only thing we could consider is adding something into the status messages when deployed without validation so it's obvious from `juju status`. I definitely feel like we need some kind of "DegradedStatus" at some point